src/arch_ppc.py

```diff
@@ -890,10 +993,25 @@ def make_memory_access(arg: Argument) -> Access:
         "fneg": lambda a: UnaryOp(op="-", expr=a.reg(1), type=Type.floatish()),
         "fmr": lambda a: a.reg(1),
         "frsp": lambda a: handle_convert(a.reg(1), Type.f32(), Type.f64()),
-        # TODO: This yields some awkward-looking C code, often in the form:
-        # `sp100 = (bitwise f64) (s32) x; y = sp104;` instead of `y = (s32) x;`.
-        # We should try to detect these idioms, along with int-to-float
-        "fctiwz": lambda a: handle_convert(a.reg(1), Type.s32(), Type.floatish()),
+        "fctiwz": lambda a: handle_convert(a.reg(1), Type.sintish(), Type.floatish()),
```

Comment:
As far as I can tell playing around on decomp.me with different compiler settings, it seems like `fctiwz` is used for either floats or doubles. ([Example scratch](https://decomp.me/scratch/Anc5r))

[This compiler-writer PDF](https://www.warthman.com/images/IBM_PPC_Compiler_Writer%27s_Guide-cwg.pdf), page 100, uses `fctiwz` for both float/double, though for double it uses a `lwa` instead of `lwz` that I'm unable to reproduce.